File: online-judge-solutions/Codeforces/1701B.py
#https://codeforces.com/problemset/problem/1701/B
#Lang		:	Python 3.8
#Time		:	374 ms
#Memory         :	12100 KB
for _ in range(int(input())):
    n=int(input())
    v,w=[i+1 for i in range(n)],[]
    for i in range(1,n+1):
        k=i
        if(k%2!=0):
            w.append(k)
            while(k*2<=n):
                w.append(k*2)
                k*=2
    print(2)
    print(*w)

File: online-judge-solutions/Codeforces/1701B.cpp
//https://codeforces.com/problemset/problem/1701/B
//Lang		:	GNU C++14
//Time		:	46 ms
//Memory	:	2400 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int t,n,j;
	cin>>t;
	while(t--){
		cin>>n;
		vector<int> v(n),w;
		for(int i=0;i<n;i++){v[i]=i+1;}
		for(int i=1;i<=n;i++){
			int k=i;
			if(k%2!=0){
				w.push_back(k);
				while(k*2<=n){
					w.push_back(k*2);
					k*=2;
				}
			}
		}
		cout<<2<<endl;
		for(int i=0;i<w.size();i++){cout<<w[i]<<" ";}cout<<endl;
	}
	return 0;
}
